alt_e2eshark/e2e_testing/test_configs/onnxconfig.py

```diff
@@ -112,3 +114,132 @@ def compile(self, mlir_module: Module, *, save_to: str = None) -> CompiledArtifa
     def run(self, artifact: CompiledArtifact, inputs: TestTensors, *, func_name="main") -> TestTensors:
         func = self.backend.load(artifact, func_name=func_name)
         return func(inputs)
+
+class CLOnnxTestConfig(TestConfig):
+    '''This is parallel to OnnxTestConfig, but uses command-line scripts for each stage.'''
+    def __init__(
+        self, log_dir: str, backend: BackendBase, torch_mlir_pipeline: Tuple[str, ...]
+    ):
+        super().__init__()
+        self.log_dir = log_dir
+        self.backend = backend
+        self.tensor_info_dict = dict()
+        if len(torch_mlir_pipeline) > 0:
+            self.pass_pipeline = "builtin.module(" + ",".join(torch_mlir_pipeline) + ")"
+        else:
+            self.pass_pipeline = None
+    
+    def import_model(self, program: OnnxModelInfo, *, save_to: str) -> Tuple[str, str]:
+        if not save_to:
+            raise ValueError("CLOnnxTestConfig requires saving artifacts")
+        # setup a detail subdirectory
+        os.makedirs(os.path.join(save_to, "detail"), exist_ok=True)
+        # setup a commands subdirectory
+        os.makedirs(os.path.join(save_to, "commands"), exist_ok=True)
+        # set file paths
+        mlir_file = os.path.join(save_to, "model.torch_onnx.mlir")
+        detail_log = os.path.join(save_to, "detail", "import_model.detail.log")
+        commands_log = os.path.join(save_to, "commands", "import_model.commands.log")
+        # get a command line script
+        script = "python -m torch_mlir.tools.import_onnx "
```

Comment:
Hey, can we possibly do this? (I haven't tried it)
<img width="344" alt="image" src="https://github.com/user-attachments/assets/a725456c-f79c-4994-b47a-6f5663e1e9c5">
Might help us avoid some python path stuff

Reply:
Ah, good point. I've been using the setup for a local torch-mlir build. Let me test out with a pip installed torch-mlir

Reply:
Running from python seems to work fine with a pip installed torch-mlir, too. When using a local build, you have to setup your pythonpath anyway, so I think it's fine. I've got a script for saving environment variables (e.g., pythonpath) to a `.env` file in the utils directory for people using local builds.

Also, I am unable to use the `torch-mlir-import-onnx` command-line tool from my local build without adding it to my path. Even worse, `torch-mlir-import-onnx` from a pip install actually throws an error:

```
Traceback (most recent call last):
  File "/home/zjgar/code/SHARK-TestSuite/alt_e2eshark/pipped.venv/bin/torch-mlir-import-onnx", line 5, in <module>
    from torch_mlir.tools.import_onnx import _cli_main
ImportError: cannot import name '_cli_main' from 'torch_mlir.tools.import_onnx' (unknown location)
```

Might be a good idea to file an issue for this...

Reply:
I'm going to merge this PR in a little bit, unless you want some time to review it further. Let me know, @saienduri .

Reply:
Looks good to me Zach, thanks for this!

Reply:
So, from the CI point of view, does it need to set any python environment variables or does the pip installed wheel work fine?

Reply:
Pip should work great!

Reply:
Just need to set the CACHE_DIR environment variable and the AZ_PRIVATE_CONNECTION (or whatever that var is called)